### ISSUE TITLE
Cancel selection on key press, then discard key press

### ISSUE THREAD

**Issue by BartXZX** [microsoft/terminal#3911]
# Description of the new feature/enhancement
Currently when cancelling a selection by pressing a key the key gets put on the command line (just like when no selection is going on). I would like to cancel selections without the key I pressed getting through to the terminal. The little selection box is bugging me when I am running some program that I don't want to provide input to, but at the same time I want to remove the selection box.

# Proposed technical implementation details (optional)

**Comment by zadjii-msft** [comment]
Yea this seems reasonable. I suppose typically I'm running `cmd` and use <kbd>esc</kbd> to dismiss the selection, but if the underlying client is also listening for escape keypresses, this would be super annoying.

**Comment by DHowett-MSFT** [comment]
We chose to specifically discard <kbd>Esc</kbd>-- we wanted to try this interaction model on for size.

I profiled a number of terminal emulators for their behavior here. Most of them just don't dismiss the selection. The ones that do also eat the keypress.

Maybe all of these implementations are wrong and we should converge on one. The reason it is the way that it is in the legacy windows Console is that QuickEdit is a separate mode that impacts the entire console interaction model (remember: edit/mark mode used to be accessible only through a menu, and fully pause the console)... but terminal doesn't have an equivalent concept.
